core/templates/dev/head/pages/exploration-editor-page/statistics-tab/services/state-improvement-suggestion.service.spec.ts

```diff
@@ -373,8 +304,16 @@ describe('StateImprovementSuggestionService', function() {
         },
       }
     };
-
-    var _createState = function(destStateName) {
+    /* eslint-disable camelcase */
```

Comment:
Can you explain why this eslint-disable is necessary? Should we be changing our eslint rules? /cc @kevinlee12

Reply:
I believe this was not intended to be a permanent state, there’s an issue for this open.

Reply:
I am not sure about why this is necessary but I added a disable since it was listed in the rules. If there is an issue open, we can remove these disable statements in the PR fixing it.

@kevinlee12 can you please provide the link to the issue? I will add the details to remove the disable statement as well.

Reply:
It’s this one: https://github.com/oppia/oppia/issues/7176

Reply:
So ... I still don't understand what is going on. What is the intended fix for this and what is blocking it? I see #7176 but I don't understand why this leads to an eslint-disable all the way over here.

(Also, I think #7176 might make a good starter project. I'll add it to the list.)

Reply:
The domain objects in backend have underscore based keys used in object definition, now these keys are passed to frontend and we use the same key names for type definition of an interface corresponding to the domain object. We have a rule in eslint which does not allow such underscore based keys. So, we either need to remove this rule or change the domain objects in the backend. So, as of now we are just using eslint-disable to disable this rule and we can remove the disable statement if we remove the camelcase rule from eslint.

What do you suggest @seanlip?

Reply:
Ah, I see, I misinterpreted the eslint-disable. Could you please change it to surround just line 310 instead of the entire thing? Initially I thought it referred to InteractionType.

I agree that this is similar in spirit to the rest of #7176 and it's fine to list it there. I thought about doing the domain object conversion in this PR but I don't think it's worth delaying the migration for this issue in particular.

Thanks!